src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+
+  hidden = false;
+
+  hide() {
+    if (this.hidden === true) {
+      this.hidden = false;
+    }
+
+    this.hidden = true;
+  }
```

Comment:
This method is misleading, as it can also unhide hidden animals (that is opposite to `hide`)
You need to either rename it or to change method logic (so it will only hide the animal)